### ISSUE TITLE
utils.set_tilt_angles does not work properly

### ISSUE THREAD

**Issue by yijiang1** [OpenChemistry/tomviz#442]
Here is a strange bug:
1. Open tomviz and load a dataset.
2. Click "Generate Tilt Series" in Tomography menu. Use default parameters.
3. When finished, the dataset is marked as a tilt series. However, its tilt angles are still zero.
<img width="282" alt="screen shot 2016-06-22 at 5 11 48 pm" src="https://cloud.githubusercontent.com/assets/13088588/16283828/78e539a6-389c-11e6-8548-2f686fcd4330.png">

**Comment by mathturtle** [comment]
I've seen this myself and tracked it down to the same root issue as #426.  But it should have been fixed by #438.  Have you tried this with a build from today?  (I know the windows build machine is a bit behind, so only the OSX builds are up to date)

**Comment by yijiang1** [comment]
> Have you tried this with a build from today?

Yes I'm running on the latest build.

**Comment by mathturtle** [comment]
I just tried the latest build and it worked fine.  Maybe the build for that merge hadn't finished when you downloaded yours?

**Comment by yijiang1** [comment]
I just tried the latest build (on Mac) and still have the problem.

**Comment by mathturtle** [comment]
So I tried it again.... re-reading the issue and doing exactly what you said.  And I was able to reproduce it.  Actually @ElliotPadgett clued me into what was going on in his point 1 on #445.  Since I thought it was a display update issue, I was re-running the transform.  And the second time, since the dataset was *already* a tilt series, it worked fine.  Then on the newer builds I was only testing the display update issue (which IS fixed).

It looks like I introduced this in #322, a case we missed when testing it.  I had wondered why reconstructions acted strange the first time I ran them on a dataset...  It turns out this is why.  I just hope my solution doesn't break anything else.

For a consistent view of the data and tilt series info, I really think we need to adopt my proposed solution to #257 where any modifications to the data source type or tilt angles are saved as operators in the pipeline.  The current code feels kludgy and has strange corner cases like this.
